pkgs/development/ocaml-modules/mdx/default.nix

```diff
@@ -0,0 +1,28 @@
+{ stdenv, fetchFromGitHub, buildDunePackage, astring, cmdliner, cppo, fmt, logs, lwt4, ocaml-migrate-parsetree, pandoc, re }:
+
+buildDunePackage rec {
+  pname = "mdx";
+  version = "1.4.0";
+
+  minimumOCamlVersion = "4.02.3";
+
+  src = fetchFromGitHub {
+    owner = "realworldocaml";
+    repo = pname;
+    rev = version;
+    sha256 = "0ljd00d261s2wf7cab086asqi39icf9zs4nylni6dldaqb027d4w";
+  };
+
+  nativeBuildInputs = [ cppo ];
+  buildInputs = [ astring cmdliner fmt logs ocaml-migrate-parsetree re ];
```

Comment:
`mdx` is being packaged because it is a dependency for `printbox` (only for the check phase), which uses the binaries. Therefore I am keeping `mdx` dependencies not propagated.

Reply:
I have unsuccessfully looked for information on when a package has dependencies on OCaml libraries they should be put in `nativeBuildInputs`, `buildInputs`, `propagatedBuildInputs`, etc.

Maybe it is worth to add a note in the OCaml section of the nixpkgs manual making it clear that when the package is intended to be used as library, its dependencies on OCaml libraries should be propagated.